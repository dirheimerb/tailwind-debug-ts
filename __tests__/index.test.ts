import postcss from 'postcss';
import tailwindcss, { Config } from 'tailwindcss';
import debugScreens from '../src/debug-screens';

import path from 'path';

const tailwindsConfig = path.resolve(__dirname, '../tailwind.config.js');

describe('debugScreens plugin', () => {
    it('should generate CSS rules for default screen', async () => {
        const config: Config = {
            plugins: [debugScreens],
            theme: {
                screens: {
                    sm: '640px',
                    md: '768px',
                    lg: '1024px',
                    xl: '1280px',
                },
            },
            content: [],
        };
        const result = await postcss(tailwindcss(config)).process('@tailwind components;', {
            from: tailwindsConfig,
        });
        expect(result.css).toMatchSnapshot();
    });

    it('should generate CSS rules for custom screen', async () => {
        const config: Config = {
            content: [],
            plugins: [debugScreens],
            theme: {
                screens: {
                    sm: '640px',
                    md: '768px',
                    lg: '1024px',
                    xl: '1280px',
                    xxl: '1600px',
                },
            },
        };
        const result = await postcss(tailwindcss(config)).process('@tailwind components;', {
            from: tailwindsConfig,
        });
        expect(result.css).toMatchSnapshot();
    });

    it('should generate CSS rules for custom screen with custom prefix', async () => {
        const config: Config = {
            content: [],
            plugins: [debugScreens],
            theme: {
                screens: {
                    sm: '640px',
                    md: '768px',
                    lg: '1024px',
                    xl: '1280px',
                    xxl: '1600px',
                },
                debugScreens: {
                    prefix: 'screen-',
                },
            },
        };
        const result = await postcss(tailwindcss(config)).process('@tailwind components;', {
            from: tailwindsConfig,
        });
        expect(result.css).toMatchSnapshot();
    });

    it('should generate CSS rules for custom screen with custom position', async () => {
        const config: Config = {
            content: [],
            plugins: [debugScreens],
            theme: {
                screens: {
                    sm: '640px',
                    md: '768px',
                    lg: '1024px',
                    xl: '1280px',
                    xxl: '1600px',
                },
                debugScreens: {
                    position: ['top', 'right'],
                },
            },
        };
        const result = await postcss(tailwindcss(config)).process('@tailwind components;', {
            from: tailwindsConfig,
        });
        expect(result.css).toMatchSnapshot();
    });
});