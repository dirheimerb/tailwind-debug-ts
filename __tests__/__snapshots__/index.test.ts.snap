// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debugScreens plugin should generate CSS rules for custom screen 1`] = `
"@media (min-width: 640px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: sm'
}
@media (min-width: 768px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: md'
}
@media (min-width: 1024px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: lg'
}
@media (min-width: 1280px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xl'
}
@media (min-width: 1600px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xxl'
}
@media (min-width: 640px; max-width: 640px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: sm'
}
@media (min-width: 768px; max-width: 768px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: md'
}
@media (min-width: 1024px; max-width: 1024px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: lg'
}
@media (min-width: 1280px; max-width: 1280px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xl'
}
@media (; ) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xxl'
}"
`;

exports[`debugScreens plugin should generate CSS rules for custom screen with custom position 1`] = `
"@media (min-width: 640px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: sm'
}
@media (min-width: 768px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: md'
}
@media (min-width: 1024px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: lg'
}
@media (min-width: 1280px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xl'
}
@media (min-width: 1600px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xxl'
}
@media (min-width: 640px; max-width: 640px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: sm'
}
@media (min-width: 768px; max-width: 768px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: md'
}
@media (min-width: 1024px; max-width: 1024px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: lg'
}
@media (min-width: 1280px; max-width: 1280px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xl'
}
@media (; ) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xxl'
}"
`;

exports[`debugScreens plugin should generate CSS rules for custom screen with custom prefix 1`] = `
"@media (min-width: 640px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-sm'
}
@media (min-width: 768px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-md'
}
@media (min-width: 1024px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-lg'
}
@media (min-width: 1280px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-xl'
}
@media (min-width: 1600px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-xxl'
}
@media (min-width: 640px; max-width: 640px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-sm'
}
@media (min-width: 768px; max-width: 768px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-md'
}
@media (min-width: 1024px; max-width: 1024px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-lg'
}
@media (min-width: 1280px; max-width: 1280px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-xl'
}
@media (; ) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen-xxl'
}"
`;

exports[`debugScreens plugin should generate CSS rules for default screen 1`] = `
"@media (min-width: 640px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: sm'
}
@media (min-width: 768px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: md'
}
@media (min-width: 1024px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: lg'
}
@media (min-width: 1280px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xl'
}
@media (min-width: 640px; max-width: 640px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: sm'
}
@media (min-width: 768px; max-width: 768px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: md'
}
@media (min-width: 1024px; max-width: 1024px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: lg'
}
@media (min-width: 1280px; max-width: 1280px) {
    position: fixed;
    z-index: 2147483647;
    -padding: .3333333em .5em;
    font-size: 12px;
    line-height: 1;
    font-family: sans-serif;
    background-color: #000;
    color: #fff;
    box-shadow: 0 0 0 1px #fff;
    content: 'screen: xl'
}"
`;
